File: Problems/No.2136_Earliest_Possible_Day_of_Full_Bloom/earliest_possible_day_of_full_bloom.cpp
class Solution {
public:
    int earliestFullBloom(vector<int>& plantTime, vector<int>& growTime) {
        
        // parameter
        std::vector<std::pair<int, int>> GP_times;
        int earliest_possible_day = 0, pT_sum = 0;

        // start computing
        for (int i=0; i<plantTime.size(); i++) GP_times.push_back(std::make_pair(growTime[i], plantTime[i]));
        std::sort(GP_times.rbegin(), GP_times.rend());
        for (auto [gT, pT]: GP_times) {

            pT_sum += pT;
            earliest_possible_day = std::max(earliest_possible_day, pT_sum+gT);

        }

        return earliest_possible_day;

    }
};

File: Problems/No.2136_Earliest_Possible_Day_of_Full_Bloom/earliest_possible_day_of_full_bloom.py
class Solution(object):
    def earliestFullBloom(self, plantTime, growTime):
        """
        :type plantTime: List[int]
        :type growTime: List[int]
        :rtype: int
        """

        # parameters
        pT_sum, earliest_possible_day = 0, 0

        # start computing
        for gT, pT in sorted(zip(growTime, plantTime), reverse=True):
            pT_sum += pT
            earliest_possible_day = max(earliest_possible_day, pT_sum+gT)

        return earliest_possible_day
